src/x11/tray_manager.cpp

```diff
@@ -67,25 +67,31 @@ tray_manager::~tray_manager() {
   deactivate();
 }
 
-void tray_manager::setup() {
+void tray_manager::setup(string tray_module_name) {
   const config& conf = config::make();
   auto bs = conf.section();
   string position;
 
   try {
     position = conf.get(bs, "tray-position");
   } catch (const key_error& err) {
-    return m_log.info("Disabling tray manager (reason: missing `tray-position`)");
   }
 
-  if (position == "left") {
+  if (!position.empty() && !tray_module_name.empty()) {
+    m_log.warn("Tray position is manually defined(`tray-position`) and also set by tray module");
+  }
+  if (position.empty() && tray_module_name.empty()) {
+    return m_log.info("Disabling tray manager (reason: tray module not defined)");
+  }
+
+  if (!tray_module_name.empty()) {
+    m_opts.adaptive = true;
```

Comment:
We now have to mutually exclusive tray settings: `adaptive` and `align`.
Whenever one is set, the other is not and they kind of control the same thing (tray position).

In my opinion it would be cleaner to have a single setting for both. For example a new `tray_pos` enum with `NONE`, `LEFT`, `CENTER`, `RIGHT`, and `MODULE`.